src/fiber/fiber.ml

```diff
@@ -394,6 +394,50 @@ module Ivar = struct
       | Empty _ -> None )
 end
 
+module Mvar = struct
+  type 'a t =
+    { writers : ('a * unit K.t) Queue.t
+    ; readers : 'a K.t Queue.t
+    ; mutable value : 'a option
+    }
+
+  let create () =
+    { value = None; writers = Queue.create (); readers = Queue.create () }
+
+  let rec step t =
```

Comment:
I don't really understand what `step` means here, which makes understanding the implementation a little bit difficult. Could you a add some comments to explain the logic of this code?

Reply:
I added some comments. Hopefully it's more clear now.